src/plot.js

```diff
@@ -143,3 +140,18 @@ function autoHeight({y, fy, fx}) {
   const ny = y ? (y.type === "ordinal" ? y.scale.domain().length : Math.max(7, 17 / nfy)) : 1;
   return !!(y || fy) * Math.max(1, Math.min(60, ny * nfy)) * 20 + !!fx * 30 + 60;
 }
+
+// Wrap the plot in a figure with a caption, if desired.
+function wrap(svg, scaleDescriptors, {caption} = {}) {
+  const scales = exposeScales(scaleDescriptors);
+  if (caption == null) {
+    return svg.scales = scales, svg;
+  }
+  const figure = create("figure");
+  figure.append(() => svg);
```

Comment:
I don’t see much value in using a D3 selection here (versus using the DOM API as we did before), since this is always just a single element.